homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,163 @@
+        """Initialize the EnOcean cover device."""
+        super().__init__(dev_id)
+        self._attr_position: int | None = None
+        self._attr_unique_id = str(combine_hex(dev_id))
+        self._attr_name = dev_name
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed or not."""
+        if self._attr_position is None:
+            return None
+        if self._attr_position > 0:
+            return False
+        return True
+
+    def value_changed(self, packet: Any) -> None:
+        """Update the internal state of the cover when a packet arrives."""
+        self._attr_position = int(packet.data[1])
+        self.schedule_update_ha_state()
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return the current position.
+
+        Avoid calibration error.
+        """
+        if self._attr_position is not None:
+            if self._attr_position <= 5:
+                return 0
+            if self._attr_position >= 95:
+                return 100
+            return self._attr_position
+
+        # ___ Force the device to reply its position.
+        # ___ The device will resend a message with its position.
+        optional = [
+            0x03,
+        ]
+        optional.extend(self.dev_id)
+        optional.extend([0xFF, 0x00])
+        self.send_command(
+            data=[0xD2, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00],
+            optional=optional,
+            packet_type=0x01,
+        )
+        return None
+
+    def open_cover(self, **kwargs: Any) -> None:
+        """Move the roller shutter up."""
+
+        # ___ The new position value will be 100.
+        # ___ Send the command to set a new position.
+        optional = [
+            0x03,
+        ]
+        optional.extend(self.dev_id)
+        optional.extend([0xFF, 0x00])
+        self.send_command(
+            data=[0xD2, 0x64, 0x7F, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00],
+            optional=optional,
+            packet_type=0x01,
+        )
+
+    def close_cover(self, **kwargs: Any) -> None:
+        """Move the roller shutter down."""
+
+        # ___ The new position value will be 0.
+        # ___ Send the command to set a new position.
+        optional = [
+            0x03,
+        ]
+        optional.extend(self.dev_id)
+        optional.extend([0xFF, 0x00])
+        self.send_command(
+            data=[0xD2, 0x00, 0x7F, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00],
+            optional=optional,
+            packet_type=0x01,
+        )
+
+    def set_cover_position(self, **kwargs: Any) -> None:
+        """Move the roller shutter to a specific position."""
+        # ___ Send the command to set a new position.
+        newVal: int = int(kwargs[ATTR_POSITION])
+        optional = [
+            0x03,
+        ]
+        optional.extend(self.dev_id)
+        optional.extend([0xFF, 0x00])
+
+        self.send_command(
+            data=[
+                0xD2,
```

Comment:
The protocol details need to be moved to a 3rd party library before we can accept this change. See previously closed PRs for this integration.